src/data/fin/vec_notation.lean

```diff
@@ -152,6 +152,11 @@ meta instance _root_.pi_fin.reflect {α : Type} [reflected α] [h : has_reflect
 | (n + 1) v := (cons_head_tail v).rec $
   (`(λ x (xs : fin n → α), vec_cons x xs).subst (h _)).subst (_root_.pi_fin.reflect _)
 
+/-- Convert a vector of pexprs to the pexpr constructing that vector.-/
+meta def _root_.pi_fin.to_pexpr : Π {n}, (fin n → pexpr) → pexpr
+| 0 v := ``(![])
+| (n + 1) v := ``(vec_cons %%(v 0) %%(_root_.pi_fin.to_pexpr $ vec_tail v))
```

Comment:
This might be appropriate as a `has_to_pexpr` instance, so that `%%v` has this meaning automatically. As of writing we only have the obvious `pexpr`, `expr`, and `reflected` instances of this typeclass; but in principle we could add instances for all "containers" that sends a container of pexprs to the pexpr that constructs that container

Reply:
I'm not comfortable enough with `has_to_pexpr` to say the instance would be unproblematic (recalling all the issues with universe polymorphism). So I'm happy to keep it as-is.